src/components/ha-base-time-input.ts

```diff
@@ -261,6 +261,7 @@ export class HaBaseTimeInput extends LitElement {
     this[textField.name] =
       textField.name === "amPm" ? textField.value : Number(textField.value);
     const value: TimeChangedEvent = {
+      days: this.enableDay ? this.days : undefined,
```

Comment:
Just a style suggestion to be consistent with the `ampm` field... either use a ternary for both or an if-block for both.

Reply:
updated as requested